pypesto/objective/amici_objective.py

```diff
@@ -223,6 +250,44 @@ def __deepcopy__(self, memodict: Dict = None) -> 'AmiciObjective':
 
         return other
 
+    def __getstate__(self) -> Dict:
+        if self.amici_object_builder is None:
+            raise NotImplementedError(
+                "AmiciObjective does not support __getstate__ without "
+                "an `amici_object_builder`.")
+
+        state = {}
+        for key in set(self.__dict__.keys()) - \
+                {'amici_model', 'amici_solver', 'edatas'}:
+            state[key] = self.__dict__[key]
+
+        amici_solver_file = tempfile.mkstemp()[1]
+        amici.writeSolverSettingsToHDF5(self.amici_solver, amici_solver_file)
+        state['amici_solver_settings'] = amici_solver_file
+
+        return state
+
+    def __setstate__(self, state: Dict):
+        if state['amici_object_builder'] is None:
+            raise NotImplementedError(
+                "AmiciObjective does not support __setstate__ without "
+                "an `amici_object_builder`.")
+
+        self.__dict__.update(state)
+
+        # note: attributes not defined in the builder are lost
+        model = self.amici_object_builder.create_model()
+        solver = self.amici_object_builder.create_solver(model)
+        edatas = self.amici_object_builder.create_edatas(model)
+
+        amici.readSolverSettingsFromHDF5(
+            state['amici_solver_settings'], solver)
+        os.remove(state['amici_solver_settings'])
```

Comment:
still worried that this limits unpickling to once.

Reply:
agreed, will be done in #278